technique/muon/run_control.py

```diff
@@ -0,0 +1,297 @@
+orient_par = "geometry"
+temp_par = "temp_label"
+field_par = "field_label"
+comment_par = "comments"
+# The key for the geometry parameter in the beamline parameters dictionary
+geometry_par = "geometry"
+
+
+def set_name_sample_par(sample_pars):
+    """
+    Ask the user if the sample parameter (name) is correct and allow them to change it if not.
+
+    Parameters
+    ----------
+    sample_pars: dict
+        The sample parameters that is to contain the name parameter.
+    """
+    if name_par in sample_pars:
+        old_sample_name = sample_pars[name_par]
+    else:
+        old_sample_name = ""
+    new_sample = get_input("Sample {}?".format(old_sample_name))
+    if new_sample != "":
+        g.change_sample_par(name_par, new_sample)
+
+
+def set_orient_sample_par(sample_pars):
+    """
+    Ask the user if the sample parameter (orientation) is correct and allow them to change it if not.
+
+    Parameters
+    ----------
+    sample_pars: dict
+        The sample parameters that is to contain the orientation parameter.
+    """
+    if orient_par in sample_pars:
+        old_orientation = sample_pars[orient_par]
+    else:
+        old_orientation = ""
+    new_orient = get_input("Orientation {}?".format(old_orientation))
+    if new_orient != "":
+        g.change_sample_par(orient_par, new_orient)
+
+
+def set_temp_sample_par(sample_pars):
+    """
+    Ask the user if the sample parameter (temp) is correct and allow them to change it if not.
+
+    Parameters
+    ----------
+    sample_pars: dict
+        The sample parameters that is to contain the temp parameter.
+    """
+    if temp_par in sample_pars:
+        old_temp = sample_pars[temp_par]
+    else:
+        old_temp = ""
+    new_temp = get_input("Temperature {}?".format(old_temp))
+    if new_temp != "":
+        g.change_sample_par(temp_par, new_temp)
+
+
+def set_field_sample_par(sample_pars):
+    """
+    Ask the user if the sample parameter (field) is correct and allow them to change it if not.
+
+    Parameters
+    ----------
+    sample_pars: dict
+        The sample parameters that is to contain the field parameter.
+    """
+    if field_par in sample_pars:
+        old_field = sample_pars[field_par]
+    else:
+        old_field = ""
+    new_field = get_input("Field {}?".format(old_field))
+    if new_field != "":
+        g.change_sample_par(field_par, new_field)
+
+
+def set_geometry_beamline_par(beamline_pars):
+    """
+    Ask the user if the sample parameter (geometry) is correct and allow them to change it if not.
+
+    Parameters
+    ----------
+    beamline_pars: dict
+        The sample parameters that is to contain the geometry parameter.
+    """
+    if geometry_par in beamline_pars:
+        old_geometry = beamline_pars[geometry_par]
+    else:
+        old_geometry = ""
+    new_geometry = get_input("Geometry (L or T) {}?".format(old_geometry))
+    if new_geometry != "":
+        g.change_beamline_par(geometry_par, new_geometry)
```

Comment:
Should this check that the input is actually one of L or T and retry if not?

Reply:
Yes, added.